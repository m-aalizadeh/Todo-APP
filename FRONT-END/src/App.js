import "./App.css";
import { Routes, Route, useNavigate } from "react-router-dom";
import Grid from "@mui/material/Grid";
import TodoList from "./components/todo";
import User from "./components/user/User";

function App() {
  const history = useNavigate();
  return (
    <Grid item xs={12}>
      <Routes>
        <Route path="/" element={<User history={history} />} />
        <Route path="/todos" element={<TodoList />} />
      </Routes>
    </Grid>
  );
}

export default App;
