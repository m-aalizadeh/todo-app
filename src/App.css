.completed {
  text-decoration: line-through;
  color: gray;
}

.App{
  margin: 10rem auto;
  width: 800px;
  display: flex;
  justify-content: center;
  flex-direction: column;
  align-items: center;
  align-content: center;

}

.todo{
  background: #3399ff;
    float: left;
    border-right: 10px;
    min-height: 500px;
    margin: 16px auto;
    border: 10px solid #eee;
    box-shadow: 0 2px 3px rgb(204, 204, 204);
    padding: 16px;
    text-align: center;
    /* background: rgb(221, 105, 63);  */
    /* float: left; */
     position: relative; 
     margin-top: 16px;
    margin-right: 16px;
    margin-left: 5px;
    width: 240px;
    max-width: 700px;
    height: max-content;
    border: 1px solid rgb(46, 40, 40);
    padding: 30px;
    text-align: center;
    border-radius: 20px;

}

p{
  background: #d633ff;
  padding: 10px;
  position: relative;
  height: 10px;
  border: 1px solid rgb(77, 19, 170);
  border-radius: 5px;
}


.todo-span{
  color: red;
  margin-left: 20px;
  padding-left: 30px;
}