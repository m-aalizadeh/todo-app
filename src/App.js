import React, { useState, useEffect } from "react";
import "./App.css";
import APIHelper from "./APIHelper.js";

function App() {
  const [todos, setTodos] = useState([]);
  const [todo, setTodo] = useState("");

  useEffect(() => {
    const fetchTodoAndSetTodos = async () => {
      const todos = await APIHelper.getAllTodos();
      console.log("before")
      console.log(todos.data);
      setTodos(todos.data);
      console.log(todos);
      console.log("after")
    };
    fetchTodoAndSetTodos();
  }, []);

  const createTodo = async e => {
    e.preventDefault();
    if (!todo) {
      alert("please enter something");
      return;
    }
    // if (todos.some(({ task }) => task === todo)) {
    //   alert(`Task: ${todo} already exists`);
    //   return;
    // }
    const newTodo = await APIHelper.createTodo(todo);
    console.log(newTodo);
    console.log(todos);
    setTodos([...todos, newTodo]);
  };

  const deleteTodo = async (e, id) => {
    try {
      e.stopPropagation();
      await APIHelper.deleteTodo(id);
      setTodos(todos.filter(({ _id: i }) => id !== i));
    } catch (err) {}
  };

  const updateTodo = async (e, id) => {
    e.stopPropagation();
    const payload = {completed: !todos.find(todo => todo._id === id).completed}
    console.log("payload: " + payload)
    const updatedTodo  = await APIHelper.updateTodo(id, payload)
    console.log("updateTodo: " + updateTodo)
    setTodos(todos.map((todo)=> todo._id === id ? updatedTodo: todo));
    
  };

  return (
    <div className="App">
      <div>
        <input
          type="text"
          value={todo}
          onChange={({ target }) => setTodo(target.value)}
          placeholder="Enter a todo"
        />
        <button type="button" onClick={createTodo}>
          Add
        </button>
      </div>

      <div className="todo">
        {todos.length ? todos.map(({ _id, task, completed }, i) => (
          <p
            key={i}
            onClick={e => updateTodo(e, _id)}
            className={completed ? "completed" : ""}
          >
            {task} 
              <span className="todo-span" onClick={e => deleteTodo(e, _id)}>DELETE</span>
          </p>
        )): <p>No Todos Yet :(</p>}
      </div>
    </div>
  );
}

export default App;