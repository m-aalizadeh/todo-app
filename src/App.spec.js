import React from 'react';
import App from './App';
import { create } from 'react-test-renderer';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';


describe("App Component", () => {
    test("Matches the snapshot", () => {
        const app = create(<App/>);
        expect(app.toJSON).toMatchSnapshot();
    });
});

describe('App', () => {
    it('renders without crashing given the required props', () => {
      const props = {
        isFetching: false,
        dispatch: jest.fn(),
        selectedSubreddit: 'reactjs',
        posts: []
      }
      const wrapper = shallow(<App {...props} />)
      expect(toJson(wrapper)).toMatchSnapshot()
    })
  })

  it('renders the Refresh button when the isFetching prop is false', () => {
    const props = {
      isFetching: false,
      dispatch: jest.fn(),
      selectedSubreddit: 'reactjs',
      posts: []
    }
     const wrapper = shallow(<App {...props} />)
     expect(wrapper.find('button').length).toBe(1);
     expect(wrapper.find('input').length).toBe(1);
  })

