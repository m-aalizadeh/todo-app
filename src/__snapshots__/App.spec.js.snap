// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component MAtches the snapshot 1`] = `[Function]`;

exports[`App Component Matches the snapshot 1`] = `[Function]`;

exports[`App renders without crashing given the required props 1`] = `
<div
  className="App"
>
  <div>
    <input
      onChange={[Function]}
      placeholder="Enter a todo"
      type="text"
      value=""
    />
    <button
      onClick={[Function]}
      type="button"
    >
      Add
    </button>
  </div>
  <ul>
    <p>
      No Todos Yet :(
    </p>
  </ul>
</div>
`;
